diff --git a/src/core/properties/volumeindicatorproperty.cpp b/src/core/properties/volumeindicatorproperty.cpp
index 07d942f15..ce9448e47 100644
--- a/src/core/properties/volumeindicatorproperty.cpp
+++ b/src/core/properties/volumeindicatorproperty.cpp
@@ -36,10 +36,10 @@ std::string VolumeIndicatorProperty::getClassIdentifier() const { return classId
 VolumeIndicatorProperty::VolumeIndicatorProperty(std::string identifier, std::string displayName,
                                                  InvalidationLevel invalidationLevel,
                                                  PropertySemantics semantics)
-    : BoolCompositeProperty(identifier, displayName, false, invalidationLevel, semantics)
-    , plane1_("plane1", "Plane 1", vec3(0.5f), vec3(0.0f, 0.0f, 1.0f))
-    , plane2_("plane2", "Plane 2", vec3(0.5f), vec3(0.0f, 1.0f, 0.0f))
-    , plane3_("plane3", "Plane 3", vec3(0.5f), vec3(1.0f, 0.0f, 0.0f)) {
+    : BoolCompositeProperty(identifier, displayName, true, invalidationLevel, semantics)
+    , plane1_("plane1", "Plane 1", vec3(0.f), vec3(0.0f, 0.0f, 1.0f))
+    , plane2_("plane2", "Plane 2", vec3(0.f), vec3(0.0f, 1.0f, 0.0f))
+    , plane3_("plane3", "Plane 3", vec3(0.f), vec3(1.0f, 0.0f, 0.0f)) {

     getBoolProperty()->setIdentifier("enable");
     getBoolProperty()->setInvalidationLevel(InvalidationLevel::InvalidResources);
