diff --git a/include/inviwo/core/properties/property.h b/include/inviwo/core/properties/property.h
index 519d0454b..b359cee3b 100644
--- a/include/inviwo/core/properties/property.h
+++ b/include/inviwo/core/properties/property.h
@@ -181,6 +181,7 @@ public:
      * @see PropertyWidget
      */
     void registerWidget(PropertyWidget* propertyWidget);
+    void initializeWidget();
 
     /**
      * Deregister a widget, the widget will no longer receive updateFromProperty calls.
diff --git a/modules/python3/bindings/src/pyproperties.cpp b/modules/python3/bindings/src/pyproperties.cpp
index c518c5b04..0e021673f 100644
--- a/modules/python3/bindings/src/pyproperties.cpp
+++ b/modules/python3/bindings/src/pyproperties.cpp
@@ -101,7 +101,11 @@ void exposeProperties(py::module &m) {
         .def("hasWidgets", &Property::hasWidgets)
         .def("setCurrentStateAsDefault", &Property::setCurrentStateAsDefault)
         .def("resetToDefaultState", &Property::resetToDefaultState)
-        .def("onChange", [](Property *p, std::function<void()> func) { p->onChange(func); });
+        .def("onChange", [](Property *p, std::function<void()> func) { p->onChange(func); })
+        .def("initializeWidget", [](Property &p){
+            p.initializeWidget();
+        })
+        ;
 
     PyPropertyClass<CompositeProperty, Property, PropertyOwner>(m, "CompositeProperty")
         .def(py::init([](const std::string &identifier, const std::string &displayName,
diff --git a/modules/qtwidgets/src/properties/isotfpropertywidgetqt.cpp b/modules/qtwidgets/src/properties/isotfpropertywidgetqt.cpp
index 36309fa61..a5c10224f 100644
--- a/modules/qtwidgets/src/properties/isotfpropertywidgetqt.cpp
+++ b/modules/qtwidgets/src/properties/isotfpropertywidgetqt.cpp
@@ -59,7 +59,7 @@ IsoTFPropertyWidgetQt::IsoTFPropertyWidgetQt(IsoTFProperty* property)
 
     setLayout(hLayout);
     updateFromProperty();
-
+    tfDialog_ = std::make_unique<TFPropertyDialog>(property);
     connect(btnOpenTF_, &IvwPushButton::clicked, [this, property]() {
         if (!tfDialog_) {
             tfDialog_ = std::make_unique<TFPropertyDialog>(property);
diff --git a/src/core/properties/property.cpp b/src/core/properties/property.cpp
index ec429e37c..b7b454230 100644
--- a/src/core/properties/property.cpp
+++ b/src/core/properties/property.cpp
@@ -161,6 +161,16 @@ void Property::registerWidget(PropertyWidget* propertyWidget) {
     propertyWidgets_.push_back(propertyWidget);
 }
 
+void Property::initializeWidget() {
+    // auto *fac2 = util::getInviwoApplication()->getPropertyWidgetFactory();
+    // fac2->create(p);
+    // auto app = util::getInviwoApplication()->getPropertyWidgetFactory();
+    // auto fac = app->getPropertyWidgetFactory();
+    auto widget = util::getInviwoApplication()->getPropertyWidgetFactory()->create(this);
+        // ->getPropertyWidgetFactory()->create(p);
+    registerWidget(widget.release());
+}
+
 void Property::deregisterWidget(PropertyWidget* propertyWidget) {
     util::erase_remove(propertyWidgets_, propertyWidget);
 }
