diff --git a/modules/fontrendering/CMakeLists.txt b/modules/fontrendering/CMakeLists.txt
index e2ffd2959..1e20507f7 100644
--- a/modules/fontrendering/CMakeLists.txt
+++ b/modules/fontrendering/CMakeLists.txt
@@ -64,7 +64,7 @@ ivw_add_to_module_pack(${CMAKE_CURRENT_SOURCE_DIR}/fonts)
 ivw_register_license_file(NAME FreeType MODULE FontRendering
     URL "https://www.freetype.org/"
     TYPE "BSD-style, acknowledgement: Portions of this software are copyright The FreeType Project (www.freetype.org). All rights reserved."
-    FILES ${CMAKE_CURRENT_SOURCE_DIR}/ext/freetype/freetype/docs/FTL.txt
+    FILES ${CMAKE_CURRENT_SOURCE_DIR}/ext/freetype/freetype/docs/FTL.TXT
 )
 
 #--------------------------------------------------------------------
